docs/release/trg-10/trg-10-02.md

```diff
@@ -0,0 +1,128 @@
+  - Scenarios written in the Given-When-Then format.
+
+### 2. Test Implementation
+
+#### 2.1. **Cucumber with Maven**
+
+- Use **Cucumber** in combination with **Maven** to generate test classes and implement them in a structured manner.
+- Ensure modular and reusable test step implementations.
+
+#### 2.2. **API Test Focus**
+
+- Prioritize API tests at the integration test level to validate end-to-end functionality.
+- Optionally include UI tests at the integration level if applicable for your software component.
+
+### 3. Automation and Integration
+
+#### 3.1. **GitHub Actions**
+
+- Integrate test execution into **GitHub Actions** workflows for automated testing.
+- Trigger tests on key events such as pull requests and merges to main branches.
+
+#### 3.2. **Test Reporting**
+
+- Generate and store detailed test reports as part of the CI/CD pipeline outputs.
+- Provide clear pass/fail results for stakeholders.
+
+### 4. Examples and Resources
+
+#### 4.1. **Code Examples**
+
+An example of such a GitHub workflow integration can be found below.
+
+```yaml
+name: Run Cucumber Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+    # Step 1: Checkout code
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    # Step 2: Set up Java
+    - name: Set up Java
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        java-version: '17'
+
+    # Step 3: Cache Maven dependencies
+    - name: Cache Maven dependencies
+      uses: actions/cache@v3
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+    # Step 4: Run Maven tests
+    - name: Run Cucumber tests
+      run: mvn clean test
+
+    # Step 5: Upload test results
+    - name: Archive test results
+      if: always()
+      uses: actions/upload-artifact@v3
+      with:
+        name: cucumber-test-results
+        path: target/surefire-reports/
+
+```
+
+#### 4.2. **Optional Scope**
+
+- The Cucumber framework allows for importing test run results from the CI/CD pipeline into various Test Management systems. Reference implementations for this are available and in use in some Eclipse Tractus-X projects. However, this is not considered part of the
+  current TRG version due Eclipse Tractus-X requirements on open source software.
+
+---
+
+## Documentation
+
+All test automation configurations should be stored in the project repository. Test results should be uploaded
+to the execution workflow as an attachment. See [code-examples](#41-code-examples).
+
+---
+
+## Compliance
+
+This TRG serves as a recommendation. While product teams may adopt other frameworks or tools to implement their test automation, the use of Cucumber with test cases written in Gherkin syntax as a basis is encouraged as a suggested best practice at the integration test level.
```

Comment:
```suggestion
the use of Cucumber with test cases written in Gherkin syntax as a basis is encouraged as a suggested best practice at the integration test level.
```

this should not be the case. 
the way tests are expressed should be a choice made by the product team based on what's the desired outcome. 
Certain products could benefit from the gherkin syntax because non-technical people could read it and understand how the product is working (please note that this is not always the case anyway :) ), but it would also be fair to point out that cucumber tests are harder to maintain because the additional ".feature to code" layer needed, and not all the teams want to deal with this burden.